drivers/hud_spm_report/app/models/hud_spm_report/fy2023/spm_enrollment.rb

```diff
@@ -0,0 +1,280 @@
+        ].inject(&:and)
+      end
+
+      nbn_cond = arel_table[:project_type].in(HudUtility2024.project_type_number_from_code(:es_nbn)).and(bed_night_cond)
+
+      ee_cond = HudSpmReport::Fy2023::SpmEnrollment.arel_table.then do |table|
+        [
+          table[:project_type].in([:es_entry_exit, :th, :ph, :sh].flat_map { |pt| HudUtility2024.project_type_number_from_code(pt) }),
+          table[:exit_date].gteq(range.begin),
+          table[:entry_date].lteq(range.end),
+        ].inject(&:and)
+      end
+
+      left_outer_joins(enrollment: :services).where(nbn_cond.or(ee_cond))
+    end
+
+    scope :literally_homeless_at_entry_in_range, ->(range) do
+      where(arel_table[:exit_date].gteq(range.begin).and(arel_table[:entry_date].lteq(range.end))).
+        where(project_type: [0, 1, 4, 8]).
+        or(where(project_type: [2, 3, 9, 10, 13], prior_living_situation: 100..199)).
+        or(where(previous_street_essh: true, project_type: [2, 3, 9, 10, 13], prior_living_situation: 200..299, length_of_stay: [2, 3])).
+        or(where(previous_street_essh: true, project_type: [2, 3, 9, 10, 13], prior_living_situation: 300..499, los_under_threshold: true))
+    end
+
+    HomelessnessInfo = Struct.new(:start_of_homelessness, :entry_date, :move_in_date, keyword_init: true)
+
+    # Unlike, most HUD reports, there is not a single enrollment per report client, so the enrollment set
+    # is constructed outside of the question universe, and then to preserve the 1:1 relationship between clients
+    # and question universe members, the question universes either refer directly to an enrollment in this set, or
+    # to an aggregation object that refers to enrollments in this set.
+    def self.create_enrollment_set(report_instance)
+      filter = ::Filters::HudFilterBase.new(user_id: User.system_user.id).update(report_instance.options)
+      enrollments = HudSpmReport::Adapters::ServiceHistoryEnrollmentFilter.new(report_instance).enrollments
+      household_infos = household(enrollments)
+      enrollments.preload(:client, :destination_client, :exit, :income_benefits, project: :funders).find_in_batches do |batch|
+        members = []
+        batch.each do |enrollment|
+          current_income_benefits = current_income_benefits(enrollment, filter.end)
+          previous_income_benefits = previous_income_benefits(enrollment, current_income_benefits&.information_date, filter.end)
+          household_info = household_infos[enrollment.household_id] ||
+            HomelessnessInfo.new(
+              start_of_homelessness: enrollment.date_to_street_essh,
+              entry_date: enrollment.entry_date,
+              move_in_date: enrollment.move_in_date,
+            ) # If there is no HoH, use the enrollment
+          members << {
+            report_instance_id: report_instance.id,
+
+            first_name: enrollment.client.first_name,
+            last_name: enrollment.client.last_name,
```

Comment:
are we not using the SpmClient model?  Seems like we could have a link here to that client instead of storing name here.  That way we could pull in other demographic data by joining SpmClient.